Puc/v4p4/UpdateChecker.php

```diff
@@ -309,6 +311,48 @@ protected function filterUpdateResult($update, $httpResult = null) {
 			return $update;
 		}
 
+		/**
+		 * The "Tested up to" field in the plugin metadata is supposed to be in the form of "major.minor",
+		 * while WordPress core's list_plugin_updates() expects the $update->tested field to be an exact
+		 * version, e.g. "major.minor.patch", to say it's compatible. In other case it shows
+		 * "Compatibility: Unknown".
+		 * The function mimics how wordpress.org API crafts the "tested" field out of "Tested up to".
+		 *
+		 * @param Puc_v4p4_Update|null $update
+		 */
+		private function fixSupportedWordpressVersion(Puc_v4p4_Update $update = null) {
+
+			if (!isset($update->tested) || !preg_match('/^\d++\.\d++$/', $update->tested)) {
+				return;
+			}
+
+			$actualWpVersions = array(); {
+
+				$wpVersion = $GLOBALS['wp_version'];
+
+				if ( function_exists('get_preferred_from_update_core') ) {
+					$coreUpdate = get_preferred_from_update_core();
+					if ( isset($coreUpdate->current) && version_compare($coreUpdate->current, $wpVersion, '>') ) {
+						$actualWpVersions[] = $coreUpdate->current;
+					}
+				}
+
+				$actualWpVersions[] = $wpVersion;
+			}
```

Comment:
This project doesn't use statement-groups in that way. Let's leave the above block without the outer `{...}` and without the extra level of indentation.